aspoet/src/main/kotlin/com/google/androidstudiopoet/models/ModuleBuildBazelBlueprint.kt

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.google.androidstudiopoet.models
+
+import com.google.androidstudiopoet.utils.joinPath
+
+class ModuleBuildBazelBlueprint(
+        additionalDependencies: Set<Dependency>,
+        override val extraLines: List<String>? = null,
+        moduleRoot: String
+) : ModuleBuildSpecificationBlueprint {
+
+    override val moduleName = moduleRoot.split("/").last()
```

Comment:
I'm not sure if this will work on Windows... You can pass it in the constructor. It is called from `ModuleBlueprint`, so you can use `AbstractModuleBlueprint.name` property there. It has the value you are looking for..